specs/agents/span-links.md

```diff
@@ -0,0 +1,36 @@
+## Span Links
+
+A Span or Transaction MAY be linked to zero or more other Spans/Transactions that are causally related.
+
+Specific use-cases for Span Links:
+
+1. When a single transaction represents the batch processing of several messages, the agent SHOULD be able to link back to the traces that have produced the messages.
+2.  When the agent receives a `traceparent` header from outside a trust boundary, it SHOULD restart the trace (creating a different trace id with its own sampling decision) and link to the originating trace.
+3. Close gap for the OTLP intake - [OTel's specification of span links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans)
+
+Spans and Transactions MUST collect links in the `links` array with the following fields on each item:
+- `trace.id`: the id of the linked trace.
+- `span.id`: the id of the linked span or transaction.
```

Comment:
I'm good with that.
Rather than changing the name of `span.id` to something like `span_or_transaction.id`, we should rename `transaction.id` to `span.id`. Transactions are just a special type of span - a span that represents an entry into a service.
But that's a discussion for another time.